source/Plugins.cpp

```diff
@@ -38,14 +64,135 @@ namespace {
 				continue;
 
 			for(const DataNode &child : node)
-				if(child.Size() == 2)
+				if(child.Size() == 3)
 				{
+					lock_guard<mutex> guard(pluginsMutex);
 					auto *plugin = plugins.Get(child.Token(0));
 					plugin->enabled = child.Value(1);
 					plugin->currentState = child.Value(1);
+					plugin->version = child.Token(2);
 				}
 		}
 	}
+
+	bool oldNetworkActivity = false;
+
+	mutex activePluginsMutex;
+	set<string> activePlugins;
+
+	// The maximum size of a plugin in bytes, this will be 1 GB.
+	const int MAX_DOWNLOAD_SIZE = 1000000000;
+
+	// Copy an entry from one archive to the other
+	void CopyData(struct archive *ar, struct archive *aw)
```

Comment:
These two functions should also either move to Files, or to a separate helper class.

Reply:
https://github.com/endless-sky/endless-sky/pull/9068/commits/8c1ec598f1050a8d74c4958ba78fcfa401411c5f#r1333472292
quyykk said different